internal/docgen/docgen.go

```diff
@@ -0,0 +1,119 @@
+package docgen
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	codacy "github.com/codacy/codacy-engine-golang-seed/v5"
+	"golang.org/x/mod/modfile"
+)
+
+const toolName string = "trivy"
+
+type DocumentationGenerator interface {
+	Generate(destinationDir string) error
+}
+
+func New() DocumentationGenerator {
+	return &documentationGenerator{}
+}
+
+type documentationGenerator struct{}
+
+func (g documentationGenerator) Generate(destinationDir string) error {
+	trivyRules := trivyRules()
+
+	trivyVersion, err := trivyVersion()
+	if err != nil {
+		return err
+	}
+
+	if err := g.createPatternsFile(trivyRules, *trivyVersion, destinationDir); err != nil {
+		return err
+	}
+
+	if err := g.createPatternsDescriptionFiles(trivyRules, destinationDir); err != nil {
+		return err
+	}
+	return nil
+}
+
+// trivyVersion returns the current version of the trivy library used.
+func trivyVersion() (*string, error) {
+	goModFilename := "go.mod"
+	dependency := "github.com/aquasecurity/trivy"
+
+	goMod, err := os.ReadFile(goModFilename)
+	if err != nil {
+		return nil, DocGenError{msg: fmt.Sprintf("Failed to load %s file", goModFilename), w: err}
+	}
+
+	file, _ := modfile.Parse(goModFilename, goMod, nil)
+	for _, r := range file.Require {
+		if r.Mod.Path == dependency {
+			version := strings.TrimPrefix(r.Mod.Version, "v")
+			return &version, nil
+		}
+	}
+	return nil, DocGenError{msg: fmt.Sprintf("%s dependency not found in %s file", dependency, goModFilename)}
+}
+
+func (g documentationGenerator) createPatternsFile(rules Rules, toolVersion, destinationDir string) error {
+	fmt.Println("Creating patterns file...")
```

Comment:
@mrfyda can we get rid of this? And the one below as well?

Reply:
I'm resolving the conversation so I can merge, but feel free to reply and we can take care of this in the next PR.